src/main/java/com/iridium/iridiumskyblock/IridiumSkyblock.java

```diff
@@ -122,7 +122,13 @@ public void onEnable() {
     private void loadDependencies() {
         PluginDependencyManager dependencyManager = SpigotDependencyManager.of(this);
         CompletableFuture<Void> loadAllDependencies = dependencyManager.loadAllDependencies();
```

Comment:
So this is being done async. What happens if we try to access a dependency while its still being downloaded? e.g. when we serialize the configs and stuff

Reply:
Although I just realised that its doing this onLoad not onEnable, but what happens if the server is down or its lagging alot and onEnable gets called before its finished?

Reply:
Yeah this needs to be called blocking anyway

Reply:
It is blocking?

Reply:
Yeah it is now

Reply:
Also, why don't you just call `.join()` instead of calling `.get()` which throws exceptions

Reply:
Ah actually I see